Hw14 spring

Caiguer opened the pull request.

Caiguer committed:
Merge branch 'hw13-di' into master

Caiguer committed:
Merge remote-tracking branch 'origin/master' into master

Caiguer committed:
Merge remote-tracking branch 'origin/master' into master

Caiguer committed:
init hibernate

Caiguer committed:
hw14 spring

stvort commented on HW14-Spring/src/main/java/com/oneso/hibernate/config/UserConfig.java:
```diff
@@ -0,0 +1,28 @@
+@Configuration
+public class UserConfig {
+
+    @Bean
+    public User initOnePerson(ServiceUserImpl serviceUser) {
+        User user = new User();
+        Address address = new Address(0, "street #123", user);
+        Phone phone = new Phone(0, "phone #1", user);
+        user.setName("John");
+        user.setAddress(address);
+        user.setPhones(List.of(phone));
+
+        serviceUser.saveUser(user);
+        return user;
+    }
```
Лучше наверное сервис создать для этих целей, а то и вовсе системой миграции воспользоваться

stvort commented on HW14-Spring/src/main/java/com/oneso/hibernate/core/service/ServiceAddressImpl.java:
```diff
@@ -0,0 +1,96 @@
+
+  private static final Logger logger = LoggerFactory.getLogger(ServiceAddressImpl.class);
+
+  private final MyCache<Long, Address> cache = new MyCache<>();
```
По заданию кеш д.б. бином, ну и в целом, чтобы иметь возможность протестировать сервисы без кеша (с его моком) лучше получать его извне

Caiguer committed:
Fix

- Test Class was removed
- UserConfig was removed
- Cache was changed to Component
